### PR TITLE
doc: Add documentation for fc-userscan

### PR DESCRIPTION
**Pull request by nichmoe**
@flyingcircusio/release-managers

## Release process

Impact:

None

Changelog:

None

## Security implications

- [ ] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
- [ ] Security requirements tested? (EVIDENCE)

### CODE REVIEW

File: doc/src/fc_userscan.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
+.. _nixos-fc_userscan:
+
+fc-userscan
+===========
+
+Protects installed packages that might still be needed from grabage collection.
```
Comment (dpausp) [inline doc/src/fc_userscan.rst]: typo in grabage

File: doc/src/fc_userscan.rst [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+Rationale
+---------
+
+As in every other linux distribution, you may install some arbitrary library, manually compile a programm
+against it and as long as this library exists, there is no problem. But NixOS some what amplities this by
+the fact that even smallest changes somewhere down the dependency chain will change checksums from which
+NixOS store paths are constructed. The garbage collection will remove the old version and break the programm.
+
+`fc-userscan <https://github.com/flyingcircusio/userscan>`_ searches for references to nix store paths in
+the service-user's directory and ensures that this packages are not garbage collected.
```
Comment (dpausp) [inline doc/src/fc_userscan.rst]: fc-userscan also runs on home dirs of real users

File: doc/src/fc_userscan.rst [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+fc-userscan
+===========
+
+Protects installed packages that might still be needed from grabage collection.
+
+Rationale
+---------
+
+As in every other linux distribution, you may install some arbitrary library, manually compile a programm
+against it and as long as this library exists, there is no problem. But NixOS some what amplities this by
```
Comment (dpausp) [inline doc/src/fc_userscan.rst]: typo in amplities

File: doc/src/fc_userscan.rst [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+the fact that even smallest changes somewhere down the dependency chain will change checksums from which
```
Comment (dpausp) [inline doc/src/fc_userscan.rst]: checksums -> hashes

File: doc/src/fc_userscan.rst [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+NixOS store paths are constructed. The garbage collection will remove the old version and break the programm.
```
Comment (dpausp) [inline doc/src/fc_userscan.rst]: will remove the old version if it's not referenced anymore

File: doc/src/fc_userscan.rst (hunk 2 above)
Comment (dpausp) [inline doc/src/fc_userscan.rst]: this -> these

File: doc/src/fc_userscan.rst [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+`fc-userscan <https://github.com/flyingcircusio/userscan>`_ searches for references to nix store paths in
+the service-user's directory and ensures that this packages are not garbage collected.
+
+Exclusion
+~~~~~~~~~
+
+Indiscriminately blocking all packages that where refered to somewhere brings a new problem.
```
Comment (dpausp) [inline doc/src/fc_userscan.rst]: "where refered to" -> are referenced

Comment (dpausp) [inline doc/src/fc_userscan.rst]: leave out "indiscriminately", it's hard to read and not really needed here

File: doc/src/fc_userscan.rst [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+For example a logfile may refere to an old and unneeded nix package and as long as this log exists
```
Comment (dpausp) [inline doc/src/fc_userscan.rst]: refere -> refer, capitalize Nix

**Review by dpausp — CHANGES_REQUESTED** [review]

**Comment by nichmoe** [comment]
Requested changes where implemented by the new commit.

**Pull request merged**